### ISSUE TITLE
test: fuzzing runner fails with outdated submodule

### ISSUE THREAD

**Issue by tsafin** [tarantool/tarantool#6979]
It was discovered the harder way that if there is function introduced in the updated submodule (i.e. `c-dt` as in #6731) then fuzzing runner is unable to properly checkout updated submodule, and still uses unapdated, thus is failing to compile the newer code which would be using api. Like in:
https://github.com/tarantool/tarantool/runs/5756635410?check_suite_focus=true#step:4:18103

```
clang-14: warning: the object size sanitizer has no effect at -O0, but is explicitly enabled: -fsanitize=array-bounds,bool,builtin,enum,float-divide-by-zero,function,integer-divide-by-zero,null,object-size,return,returns-nonnull-attribute,shift,signed-integer-overflow,unsigned-integer-overflow,unreachable,vla-bound,vptr [-Winvalid-command-line-argument]
/src/tarantool/src/lib/core/datetime.c:100:7: error: implicit declaration of function 'dt_from_ymd_checked' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                if (dt_from_ymd_checked(year + 1900, mon + 1, mday, &dt) == false)
                    ^
1 error generated.
```
or https://github.com/tarantool/tarantool/runs/5756635172?check_suite_focus=true

But this `dt_from_ymd_checked` is defined in the `c-dt/dt_core.h` iff `DT_PARSE_ISO_TNT` was defined. (And it is defined in the header part of `datetime.c`)

Further observations revealed that all runners which succeeded compiling this pull-request were checking out `c-dt` commit 
```
2022-03-29T11:02:45.9790631Z Submodule path 'third_party/c-ares': checked out 'bbbffa4da8baf35b0e4e1c376e38018f9a8bcb4e'
2022-03-29T11:02:46.0163868Z Submodule path 'third_party/c-dt': checked out 'cec6acebb54d9e73ea0b99c63898732abd7683a6'
2022-03-29T11:02:46.0800553Z Submodule path 'third_party/curl': checked out '3266b35bbe21c68dea0dc7ccd991eb028e6d360c'
```
while troublesome log showed:
```
2022-03-28T13:48:23.5541929Z Submodule path 'third_party/c-ares': checked out 'bbbffa4da8baf35b0e4e1c376e38018f9a8bcb4e'
2022-03-28T13:48:23.5790098Z Submodule path 'third_party/c-dt': checked out 'cbb3fc27c104aa7703b01a4108ce7871e1a28a1c'
2022-03-28T13:48:23.8617596Z Submodule path 'third_party/curl': checked out '3266b35bbe21c68dea0dc7ccd991eb028e6d360c'
```

and cbb3fc27c104aa7703b01a4108ce7871e1a28a1c is 2 commits behind cec6acebb54d9e73ea0b99c63898732abd7683a6

```
01:53 $ git log -3 cec6acebb54d9e73ea0b99c63898732abd7683a6
commit cec6acebb54d9e73ea0b99c63898732abd7683a6 (origin/tarantool-master-v1)
Author: Timur Safin <tsafin@tarantool.org>
Date:   Thu Jan 27 00:27:38 2022 +0300

    build: add simple github action

commit e6214325fe8d4336464ebae859ac2b456fd22b77
Author: Timur Safin <tsafin@tarantool.org>
Date:   Wed Jan 26 12:42:36 2022 +0300

    build, datetime: Tarantool extension to iso8601
...
    This code is not enabled by default and only activated by definition of
    DT_PARSE_ISO_TNT macro (and DT_PARSE_ISO_YEAR0).

...
commit cbb3fc27c104aa7703b01a4108ce7871e1a28a1c (HEAD)
Author: Timur Safin <tsafin@tarantool.org>
Date:   Sat Apr 24 02:01:20 2021 +0300

    build: Tarantool CMake integration

...
```
[tarantool-build-step.3-centos7-ok.log](https://github.com/tarantool/tarantool/files/8384645/tarantool-build-step.3-centos7-ok.log)
[tarantool-build-step.3-fuzzer-not.ok.log](https://github.com/tarantool/tarantool/files/8384647/tarantool-build-step.3-fuzzer-not.ok.log)

Unfortunately, I've not found any obvious, relevant knob in fuzzing workflow, which might fix such outdated submodule checkout. Please point me to any if I've missed one. @ligurio ?

**Comment by ylobankov** [comment]
Submitted a fix https://github.com/google/oss-fuzz/issues/7500

**Comment by ylobankov** [comment]
> Submitted the fix [google/oss-fuzz#7500](https://github.com/google/oss-fuzz/pull/7500)

The fix is merged. So this issue can be closed.
